### ISSUE TITLE
EasyMock::spy() fails to assert that the method is actually called at least once

### ISSUE THREAD

**Issue by jdreesen** [mnapoli/phpunit-easymock#1]
While refactoring the factory definition tests for PHP-DI I noticed that `EasyMock::spy()` fails to assert that the method is actually called at least once.

Sadly it seems I don't know how to write a (failing) test for this behaviour, even if I just tried for about half an our :( 

Although [the current test does nothing](https://github.com/mnapoli/phpunit-easymock/blob/master/tests%2FMockClassTest.php?ts=2#L109) it should actually fail, but it doesn't.

To reproduce this you can try the following:

``` php
// Should throw "Expected invocation at least once but it never occured", but it doesn't.
EasyMock::spy('My\Class', [
    'sayHello' => 'Hello',
]);
```

``` php
// Throws exception with message: "Expected invocation at least once but it never occured"
$this->getMock('My\Class')
            ->expects($this->atLeastOnce())
            ->method('sayHello');
```

**Comment by mnapoli** [comment]
Damn!

For a failing test this should be enough I guess:

``` php
    /**
     * @test
     */
    public function should_allow_to_spy_method_calls_1()
    {
        /** @var ClassFixture $mock */
        $mock = EasyMock::spy('EasyMock\Test\Fixture\ClassFixture', array(
            'foo' => 'bar',
        ));
        $this->assertEquals('bar', $mock->foo());
    }

    /**
     * @test
     * @expectedException \PHPUnit_Framework_ExpectationFailedException
     * @expectedExceptionMessage Expected invocation at least once but it never occured
     */
    public function should_allow_to_spy_method_calls_2()
    {
        EasyMock::spy('EasyMock\Test\Fixture\ClassFixture', array(
            'foo' => 'bar',
        ));
    }
```

I guess this is due because this is missing: https://github.com/sebastianbergmann/phpunit/blob/master/src/Framework/TestCase.php#L1303

With PHPUnit, `getMock()` will register the mock. With EasyMock that doesn't happen… That probably means a big change in how this library works, maybe by providing a new base test class to extend instead of a static `EasyMock` class…

If you have other ideas? I'll have another look later today (including your other PR :smiley:)

**Comment by mnapoli** [comment]
Fixed by #2 thanks @jdreesen!
